appservice/src/createAppService/AppInsightsLocationStep.ts

```diff
@@ -0,0 +1,128 @@
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { nonNullProp } from "../utils/nonNull";
+import { requestUtils } from "../utils/requestUtils";
+import { IAppServiceWizardContext } from "./IAppServiceWizardContext";
+
+export class AppInsightsLocationStep extends AzureWizardPromptStep<IAppServiceWizardContext> {
+
+    public async prompt(wizardContext: IAppServiceWizardContext): Promise<void> {
+        const options: IAzureQuickPickOptions = { placeHolder: localize('locationAppInsights', 'Select a location for the new App Insights component.'), id: `AppInsightsLocationStep/${wizardContext.subscriptionId}` };
+        wizardContext.newAppInsightsLocation = (await ext.ui.showQuickPick(this.getQuickPicks(wizardContext), options)).label;
+    }
+
+    public shouldPrompt(wizardContext: IAppServiceWizardContext): boolean {
+        return !wizardContext.appInsightsComponent && !wizardContext.newAppInsightsLocation;
+    }
+
+    public async getSupportedLocation(wizardContext: IAppServiceWizardContext, location: Location): Promise<string | undefined> {
+        // tslint:disable-next-line: strict-boolean-expressions
+        const locations: Location[] = await this.getLocations(wizardContext) || [];
+        const locationName: string = nonNullProp(location, 'name');
+        // still need to do a name step
+        if (locations.some((loc) => loc === location.displayName)) {
+            wizardContext.telemetry.properties.locationSupported = 'true';
+            return locationName;
+        } else {
+            // If there is no exact match, then query the regionMapping.json
+            const pairedRegions: string[] | undefined = await this.getPairedRegions(locationName);
+            if (pairedRegions.length > 0) {
+                // if there is at least one region listed, return the first
+                wizardContext.telemetry.properties.locationSupported = 'pairedRegion';
+                return pairedRegions[0];
+            }
+
+            wizardContext.telemetry.properties.locationSupported = 'false';
+            return undefined;
+        }
+    }
+
+    public async getPairedRegions(locationName: string): Promise<string[]> {
+        try {
+            const request: requestUtils.Request = await requestUtils.getDefaultRequest('https://appinsights.azureedge.net/portal/regionMapping.json');
+            const response: string = await requestUtils.sendRequest(request);
+            const regionMappingJson: RegionMappingJsonResponse = <RegionMappingJsonResponse>JSON.parse(response);
+
+            // tslint:disable-next-line: strict-boolean-expressions
+            if (regionMappingJson.regions[locationName]) {
+                return regionMappingJson.regions[locationName].pairedRegions;
+            }
+        } catch (error) {
+            // ignore the error
+        }
+        return [];
+    }
+
+    private generalizeLocationName(name: string | undefined): string {
```

Comment:
Copying code I see?

Honestly I'm really not a fan of this entire file. I feel like you should be changing LocationListStep to fit your purposes, not pulling bits and pieces into an entirely new file that only helps in one scenario. For example, it seems like you created this file for a few reasons:
1. You wanted to prompt for location twice
1. You wanted to put recommended locations at the top of the list

Both of those are general things that could be supported for many scenarios outside of App Insights. You're essentially taking a shortcut and not helping those future scenarios.